state_machine/applications/system-check/tests/logging_infrastructure_test.py

```diff
@@ -0,0 +1,44 @@
+from lib.logging import get_logfile_name, clear_storage, log
+from os import listdir
+
+
+def run(hardware_dict, result_dict):
+
+    # if no SD Card detected, update result dictionary and return
+    if not hardware_dict["SDcard"]:
+        result_dict["LoggingInfrastructure_Test"] = (
+            "Cannot test Logging Infrastructure; no SD Card detected", None)
+        return result_dict
+
+    # else, clear storage and run test
+    print("Starting logging infrastructure test...")
+    clear_storage()
+    print(f"/sd/ directory: {listdir('/sd/')}")
+
+    for i in range(18):
+        msg = 'x' * 100
+        log(msg, 1000)
+
+    logfile1 = get_logfile_name(0)
+    logfile2 = get_logfile_name(1)
+
+    # should create 2 files
```

Comment:
Does running listdir create a file???

Reply:
no it doesn't -- the comment is just poorly worded. We want to check that 2 files were created by making sure the filename is in listdir. exists(filename) isn't supported by circuitpython, so this is how i'm checking that 2 files were created

I'll fix the wording here, thanks for bringing it up